ir/value.cpp

```diff
@@ -110,4 +111,37 @@ expr Input::getTyVar() const {
   return expr::mkVar(tyname.c_str(), 2);
 }
 
+
+static string genVectorName(const std::vector<Value *> &vals) {
+  ostringstream ss;
+  ss << "<";
+  for (unsigned idx = 0; idx < vals.size(); ++idx) {
+    ss << *vals[idx];
+    if (idx != vals.size() -1)
```

Comment:
it's easier to check for idx != 0 and pre-add a comma

Reply:
fixed :)